test-case/multiple-pause-resume.sh

```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+
+##
+## Case Name: Run multiple pipeline for pause resume
+## Preconditions:
+##    N/A
+## Description:
+##    pickup multiple pipline to do pause resume
+##    fake pause/resume with expect
+##    expect sleep for sleep time then mocks spacebar keypresses ' ' to
+##    cause resume action
+## Case step:
+##    1. run 1st pipeline 
+##    2. pickup any other pipline
+##    3. use expect to fake pause/resume in each pipeline
+##    4. go through with tplg file
+## Expect result:
+##    no errors occur for either process
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=3
+
+OPT_OPT_lst['c']='count'    OPT_DESC_lst['c']='test pipeline count'
+OPT_PARM_lst['c']=1         OPT_VALUE_lst['c']=2
+
+OPT_OPT_lst['r']='loop'     OPT_DESC_lst['r']='pause resume repeat count'
+OPT_PARM_lst['r']=1         OPT_VALUE_lst['r']=3
+
+OPT_OPT_lst['i']='min'      OPT_DESC_lst['i']='random range min value, unit is ms'
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']='100'
+
+OPT_OPT_lst['a']='max'      OPT_DESC_lst['a']='random range max value, unit is ms'
+OPT_PARM_lst['a']=1         OPT_VALUE_lst['a']='200'
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+repeat_count=${OPT_VALUE_lst['r']}
+loop_count=${OPT_VALUE_lst['l']}
+# configure random value range
+rnd_min=${OPT_VALUE_lst['i']}
+rnd_max=${OPT_VALUE_lst['a']}
+rnd_range=$[ $rnd_max - $rnd_min ]
+[[ $rnd_range -le 0 ]] && dlogw "Error random range scope [ min:$rnd_min - max:$rnd_max ]" && exit 2
+
+tplg=${OPT_VALUE_lst['t']}
+func_pipeline_export $tplg
+# get the min value of TPLG:'pipeline count' with Case:'pipeline count'
+[[ $PIPELINE_COUNT -gt ${OPT_VALUE_lst['c']} ]] && max_count=${OPT_VALUE_lst['c']} || max_count=$PIPELINE_COUNT
```

Comment:
max_count should be compared to the number of the pcms ,not the pipeline.

Reply:
we have some blacklist of the pipeline, so here if compare the pcms, it will include it